src/node/rpc/member_frontend.h

```diff
@@ -2110,6 +2241,533 @@ namespace ccf
+          {
+            return make_error(
+              HTTP_STATUS_INTERNAL_SERVER_ERROR,
+              ccf::errors::InternalError,
+              "Validation failed to return an object");
+          }
+
+          std::string description;
+          auto desc = context(JS_GetPropertyStr(context, val, "description"));
+          if (JS_IsString(desc))
+          {
+            auto cstr = JS_ToCString(context, desc);
+            description = std::string(cstr);
+            JS_FreeCString(context, cstr);
+          }
+
+          auto valid = context(JS_GetPropertyStr(context, val, "valid"));
+          if (!JS_ToBool(context, valid))
+          {
+            return make_error(
+              HTTP_STATUS_BAD_REQUEST,
+              ccf::errors::ProposalFailedToValidate,
+              fmt::format("Proposal failed to validate: {}", description));
+          }
+
+          auto pm =
+            ctx.tx.rw<ccf::jsgov::ProposalMap>("public:ccf.gov.proposals.js");
+          // Introduce a read dependency, so that if identical proposal
+          // creations are in-flight and reading at the same version, all except
+          // the first conflict and are re-executed. If we ever produce a
+          // proposal ID which already exists, we must have a hash collision.
+          if (pm->has(proposal_id))
+          {
+            return make_error(
+              HTTP_STATUS_INTERNAL_SERVER_ERROR,
+              ccf::errors::InternalError,
+              "Proposal ID collision.");
+          }
+          std::string proposal_string(body, body_len);
+          pm->put(proposal_id, proposal_string);
+
+          auto pi = ctx.tx.rw<ccf::jsgov::ProposalInfoMap>(
+            "public:ccf.gov.proposals_info.js");
+          pi->put(
+            proposal_id,
+            {caller_identity.member_id, ccf::ProposalState::OPEN, {}});
+
+          record_voting_history(
+            ctx.tx, caller_identity.member_id, caller_identity.signed_request);
+
+          auto rv = resolve_proposal(
+            ctx.tx, proposal_id, proposal_string, constitution.value());
+          pi->put(proposal_id, {caller_identity.member_id, rv.state, {}});
+          return make_success(rv);
+        };
+
+      make_endpoint(
+        "proposals.js",
+        HTTP_POST,
+        json_adapter(post_proposals_js),
+        member_sig_only)
+        .set_auto_schema<jsgov::Proposal, jsgov::ProposalInfo>()
+        .install();
+
+      auto get_proposal_js =
+        [this](endpoints::ReadOnlyEndpointContext& ctx, nlohmann::json&&) {
+          const auto& caller_identity =
+            ctx.get_caller<ccf::MemberSignatureAuthnIdentity>();
+          if (!check_member_active(ctx.tx, caller_identity.member_id))
+          {
+            return make_error(
+              HTTP_STATUS_FORBIDDEN,
+              ccf::errors::AuthorizationFailed,
+              "Member is not active.");
+          }
+
+          // Take expand=ballots, return eg. "ballots": 3 if not set
```

Comment:
Never use the same field for different things. Better to name it `"ballot_count"` or similar.

Reply:
Ok